internal/auth/testing.go

```diff
@@ -0,0 +1,55 @@
+package auth
+
+import (
+	"context"
+	"sort"
+	"testing"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/db/timestamp"
+	"github.com/stretchr/testify/require"
+)
+
+// ManagedGroupMemberAccount represents an entry from
+// auth_managed_group_member_account.  These are used to determine the account
+// ids where are a member of managed groups.  See: oidc and ldap managed groups
+// as well as iam role grants.
+type ManagedGroupMemberAccount struct {
+	CreateTime     *timestamp.Timestamp
+	MemberId       string
+	ManagedGroupId string
+	tableName      string
+}
+
+// SetTableName sets the table name.
+func (a *ManagedGroupMemberAccount) SetTableName(n string) {
+	a.tableName = n
+}
+
+// TableName returns the table name.
+func (a *ManagedGroupMemberAccount) TableName() string {
+	if a.tableName != "" {
+		return a.tableName
+	}
+	return "auth_managed_group_member_account"
+}
+
+// TestSortManagedGroupMemberAccounts simply sorts them by public id to make
+// comparisons a bit easier.
+func TestSortManagedGroupMemberAccounts(t testing.TB, m []*ManagedGroupMemberAccount) {
+	sort.Slice(m, func(a, b int) bool {
+		return m[a].MemberId < m[b].MemberId
+	})
```

Comment:
Only noting this because you've used the `slices` package in other places, maybe you want https://pkg.go.dev/golang.org/x/exp/slices#Sort?

Reply:
the first clone is needed to ensure that `mg is not changed` (part of the contract) before we add the public id.    This clone is needed to ensure that retried transactions have an changed copy to work on (since create could change it).

Reply:
We're using `sort.Slice(...)` in 14 places already in this pkg.     I think, slice.SortFunc(...) would be the equivalent and I'm not sure why I'd chose one over the other.... and I don't think any of our "sort" use cases warrant benchmarks to prove which is better.   For now, it's more consistent to stick with sort.Slice but in all honesty is "potato" vs "potato" for now until there's benchmark data.

Reply:
Totally fair, I will just note that I think `slices.Sort` is strictly better because it provides static type safety while `sort.Slice` has to accept an `any`. You also don't need to specify a sort function if you're using the basic comparison of the type. I don't know about any performance differences. Definitely worth staying consistent in this package though 👍🏻.